### PR TITLE
Make tests more robust against weird pathname/libuv states.

### LINKED ISSUE #15878: Centos 7 buildbot frequently failing reflection and ccall tests
https://build.julialang.org/builders/build_centos7.1-x64?numbuilds=250

e.g. https://build.julialang.org/builders/build_centos7.1-x64/builds/693/steps/shell_2/logs/stdio (ccall failure doesn't happen every time, reflection failure has been)
```
	From worker 8:	     * ccall                Test Failed
	From worker 8:	  Expression: v < 10 ^ n
	From worker 8:	   Evaluated: 1000 < 1000
[snip]
	From worker 10:	     * reflection           Test Failed
	From worker 10:	  Expression: !(contains(str,"_$(i)"))
[snip]
Exception running test ccall :
On worker 8:
LoadError: There was an error during testing
 in record at ./test.jl:322
 in do_test at ./test.jl:220
 [inlined code] from ./test.jl:137
 in anonymous at ./<no file>:4294967295
 [inlined code] from ./essentials.jl:82
 in include_string at ./loading.jl:371
 in include_from_node1 at ./loading.jl:420
 [inlined code] from ./util.jl:179
 in runtests at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/testdefs.jl:7
 in #16 at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl:36
 in run_work_thunk at ./multi.jl:744
 [inlined code] from ./multi.jl:1038
 in #251 at ./event.jl:46
while loading /home/centos/buildbot/slave/build_centos7_1-x64/build/test/ccall.jl, in expression starting on line 506
Exception running test reflection :
On worker 10:
LoadError: There was an error during testing
 in record at ./test.jl:322
 in do_test at ./test.jl:220
 [inlined code] from ./test.jl:137
 in test_typed_ast_printing at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/reflection.jl:353
 [inlined code] from ./essentials.jl:82
 in include_string at ./loading.jl:371
 in include_from_node1 at ./loading.jl:420
 [inlined code] from ./util.jl:179
 in runtests at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/testdefs.jl:7
 in #16 at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl:36
 in run_work_thunk at ./multi.jl:744
 [inlined code] from ./multi.jl:1038
 in #251 at ./event.jl:46
while loading /home/centos/buildbot/slave/build_centos7_1-x64/build/test/reflection.jl, in expression starting on line 371
ERROR: LoadError: Some tests exited with errors.
 in (::##11#19)() at /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl:64
 in cd(::##11#19, ::ASCIIString) at ./file.jl:48
 in include(::ASCIIString) at ./boot.jl:234
 in include_from_node1(::UTF8String) at ./loading.jl:417
 in process_options(::Base.JLOptions) at ./client.jl:263
 in _start() at ./client.jl:319
while loading /home/centos/buildbot/slave/build_centos7_1-x64/build/test/runtests.jl, in expression starting on line 13
```

### PR DESCRIPTION
**Pull request by yuyichao**
Fixes #15878

### CODE REVIEW

**Comment by yuyichao** [comment]
Relevant tests passed on the problematic buildbot http://buildbot.e.ip.saba.us:8010/builders/build_centos7.1-x64/builds/745/steps/shell_2/logs/stdio

File: test/ccall.jl [hunk 1]
```diff
@@ -506,19 +506,20 @@ threadcall_test_func(x) =
...
-    d=Dict()
+    # Count the original thread too instead of relying on libuv to have
+    # more than one active worker threads.
+    d = Dict(ccall(:uv_thread_self, Culong, ())=>1)
     for t in tids
         d[t] = get(d, t, 0) + 1
     end
 
-    @test sum(values(d)) == 10^n
+    @test sum(values(d)) == 10^n + 1
```
Comment (amitmurthy) [inline test/ccall.jl]: Add a test to make sure that `libuv` is actually executing the `@threadcall` in another thread?

`@test length(d) > 1`

Comment (yuyichao) [inline test/ccall.jl]: Isn't that the point of the `<=` test below?

Comment (amitmurthy) [inline test/ccall.jl]: You are right. Sorry about that.

Comment (yuyichao) [inline test/ccall.jl]: I updated the test to only check if the function is not running on the original, which I think is the only thing we care about here.

Comment (amitmurthy) [inline test/ccall.jl]: OK.

Comment (tkelman) [inline test/ccall.jl]: ah whoops, should have just reset my commit out, sorry bout that

**Pull request merged**